service/src/test/java/uk/gov/hmcts/reform/fpl/service/CafcassEmailContentProviderSDOIssuedTest.java

```diff
@@ -0,0 +1,109 @@
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.springframework.test.context.junit.jupiter.SpringExtension;
+import uk.gov.hmcts.reform.fpl.config.CafcassLookupConfiguration;
+import uk.gov.hmcts.reform.fpl.config.CafcassLookupConfiguration.Cafcass;
+import uk.gov.hmcts.reform.fpl.config.LocalAuthorityNameLookupConfiguration;
+import uk.gov.hmcts.reform.fpl.model.CaseData;
+import uk.gov.hmcts.reform.fpl.model.HearingBooking;
+import uk.gov.hmcts.reform.fpl.model.Respondent;
+import uk.gov.hmcts.reform.fpl.model.RespondentParty;
+import uk.gov.hmcts.reform.fpl.model.common.Element;
+import uk.gov.hmcts.reform.fpl.service.email.content.CafcassEmailContentProvider;
+
+import java.io.IOException;
+import java.time.LocalDate;
+import java.util.Map;
+import java.util.UUID;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.BDDMockito.given;
+import static uk.gov.hmcts.reform.fpl.CaseDefinitionConstants.CASE_TYPE;
+import static uk.gov.hmcts.reform.fpl.CaseDefinitionConstants.JURISDICTION;
+import static uk.gov.hmcts.reform.fpl.utils.CoreCaseDataStoreLoader.populatedCaseDetails;
+
+@ExtendWith(SpringExtension.class)
+class CafcassEmailContentProviderSDOIssuedTest {
+
+    private static final String LOCAL_AUTHORITY_CODE = "example";
+    private static final String CAFCASS_NAME = "Test cafcass";
+    private static final String COURT_EMAIL_ADDRESS = "FamilyPublicLaw+test@gmail.com";
+
+    @Mock
+    private CafcassLookupConfiguration cafcassLookupConfiguration;
+
+    @Mock
+    private LocalAuthorityNameLookupConfiguration localAuthorityNameLookupConfiguration;
+
+    @Mock
+    MapperService mapperService;
+
+    @Mock
+    DateFormatterService dateFormatterService;
+
+    @Mock
+    HearingBookingService hearingBookingService;
+
+    @InjectMocks
+    private CafcassEmailContentProvider cafcassEmailContentProvider;
+
+    @BeforeEach
+    void setup() {
+        given(mapperService.mapObject(Mockito.any(), Mockito.any()))
+            .willReturn(CaseData.builder().familyManCaseNumber("12345").respondents1(ImmutableList.of(
+                Element.<Respondent>builder()
+                    .value(Respondent.builder()
+                        .party(RespondentParty.builder()
+                            .lastName("Moley")
+                            .build())
+                        .build())
+                    .build()))
+                .hearingDetails(ImmutableList.of(
+                    Element.<HearingBooking>builder()
+                        .id(UUID.randomUUID())
+                        .value(HearingBooking.builder().date(LocalDate.of(2020,10, 27)).build())
+                        .build())).build());
+
+        given(hearingBookingService.getMostUrgentHearingBooking(Mockito.any())).willReturn(HearingBooking.builder()
+            .date(LocalDate.of(2020,10,27)).build());
+
+        given(dateFormatterService.formatLocalDateToString(Mockito.any(),Mockito.any()))
```

Comment:
don't think I am a fan of these any methods, sometimes it makes sense, here I am not convinced 🤔

Reply:
To be honest I think here we could test the real thing. Just use @Spy annotation and create new DateFormatterService()